sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java

```diff
@@ -140,15 +140,17 @@ private static void validatePubsubName(String name) {
    * Populate common {@link DisplayData} between Pubsub source and sink.
    */
   private static void populateCommonDisplayData(DisplayData.Builder builder,
-      String timestampLabel, String idLabel, String topic) {
+      String timestampLabel, String idLabel, ValueProvider<PubsubTopic> topic) {
     builder
         .addIfNotNull(DisplayData.item("timestampLabel", timestampLabel)
             .withLabel("Timestamp Label Attribute"))
         .addIfNotNull(DisplayData.item("idLabel", idLabel)
             .withLabel("ID Label Attribute"));
 
     if (topic != null) {
-      builder.add(DisplayData.item("topic", topic)
+      String topicString = topic.isAccessible() ? topic.get().asPath()
+          : topic.toString();
+      builder.add(DisplayData.item("topic", topicString)
```

Comment:
Answered: it only accepts VP<KnownType> https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java#L880